x-pack/plugins/security_solution/public/common/hooks/eql/api.ts

```diff
@@ -3,11 +3,21 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { Unit } from '@elastic/datemath';
 
 import { HttpStart } from '../../../../../../../src/core/public';
 import { DETECTION_ENGINE_EQL_VALIDATION_URL } from '../../../../common/constants';
 import { EqlValidationSchema as EqlValidationRequest } from '../../../../common/detection_engine/schemas/request/eql_validation_schema';
 import { EqlValidationSchema as EqlValidationResponse } from '../../../../common/detection_engine/schemas/response/eql_validation_schema';
+import { DataPublicPluginStart } from '../../../../../../../src/plugins/data/public';
+import {
+  EqlSearchStrategyRequest,
+  EqlSearchStrategyResponse,
+} from '../../../../../data_enhanced/common';
+import { getEqlAggsData, getSequenceAggs } from './helpers';
+import { EqlPreviewResponse, Source } from './types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { EqlSearchResponse } from '../../../../server/lib/types';
```

Comment:
The linter error I think is very valid. It's letting us know that we are using Eql types that should be within our types under the shared folder. We run a high risk of pulling things we don't want from the server or getting some server code from a module running expectedly.

We really should instead move this to the common side and use the type from there if possible.

Reply:
Addressed it here: https://github.com/elastic/kibana/pull/78985/commits/ece38ef4fc2918b570731bf6c6a34784e992d5d8

I only moved what was absolutely necessary to reduce number of changes.